moabb/paradigms/motor_imagery.py

```diff
@@ -82,43 +85,109 @@ def scoring(self):
-    datasets to have exactly n classes. Uses 'accuracy' as metric
+class MotorImageryMultiPass(BaseMotorImagery):
 
-    """
-
-    def __init__(self, n_classes, **kwargs):
-        self.n_classes = n_classes
+    def __init__(self, fbands=np.array([[8, 14], [20, 30]]),
+                 channels=None, **kwargs):
+        """init"""
         super().__init__(**kwargs)
+        self.fbands = fbands
+        self.channels = channels
 
-    def verify(self, d):
-        print('Warning: Assumes events have already been selected per dataset')
-        super().verify(d)
-        assert len(d.selected_events) == self.n_classes
-
-
-class LeftRightImagery(BaseMotorImagery):
-    """Motor Imagery for left hand/right hand classification
-
-    Metric is 'roc_auc'
-
-    """
 
-    def verify(self, d):
-        events = ['left_hand', 'right_hand']
-        super().verify(d)
-        assert set(events) <= set(d.event_id.keys())
-        d.selected_events = dict(zip(events, [d.event_id[s] for s in events]))
+    def _epochs(self, raws, event_dict, time):
+        '''Take list of raws and returns a list of epoch objects. Implements
+        imagery-specific processing as well
 
-    @property
-    def scoring(self):
-        return 'roc_auc'
+        '''
+        if type(raws) is not list:
+            raws = [raws]
+        ep = []
+        out_dict = {i:[] for i in range(self.fbands.shape[0])}
+        for raw in raws:
+            events = mne.find_events(raw, shortest_event=0, verbose=False)
+            channels = () if self.channels is None else self.channels
+            # picks channels
+            picks = mne.pick_types(raw.info, eeg=True, stim=False,
+                                   include=channels)
 
-    @property
-    def datasets(self):
-        return utils.dataset_search(paradigm='imagery',
-                                    events=['right_hand', 'left_hand'],
-                                    has_all_events=True)
+            # ensure events are desired:
+            if len(events) > 0:
+                keep_events = {key: val for key, val in event_dict.items() if
+                               val in np.unique(events[:, 2])}
+                if len(keep_events) > 0:
+                    # copy before filtering, so we let raws intact.
+                    for band_ind, (bp_low, bp_high) in enumerate(self.fbands):
+                        raw_f = raw.copy().filter(bp_low, bp_high, method='iir',
+                                                picks=picks, verbose=False)
+                        epochs = mne.Epochs(raw_f, events, event_id=keep_events,
+                                            tmin=time[0], tmax=time[1], proj=False,
+                                            baseline=None, preload=True,
+                                            verbose=False, picks=picks)
+                        out_dict[band_ind].append(epochs)
+
+        return out_dict
```

Comment:
instead of a list of epochs, a dict of indices: lists for each band (yes it could have simply been another list but it's like 4 lists deep already...).This may have been a stupid design choice and we should just switch to lists of lists of epochs